src/main/scala/com/github/traviscrawford/spark/dynamodb/ItemConverter.scala

```diff
@@ -14,24 +14,25 @@ private[dynamodb] object ItemConverter {
     * Note this is a simple converter for use in tests while determining the overall structure,
     * and does not focus on efficiency or handling all data types at this time.
     */
-  def toRow(item: Item, schema: StructType): Row = {
+  def toRow(item: Item, schema: StructType, requiredColumns: Seq[String]): Row = {
```

Comment:
This change looks good - since we implement `PrunedScan` it's definitely better to return just the necessary columns. Can you open a separate PR with this change?